lib/figma.js

```diff
@@ -45,3 +45,100 @@ exports.findAllTokens = (data, tokenType) =>
         .toLowerCase()
         .includes(tokenType.toLowerCase()),
   );
+
+/**
+ * Parses a flattened Figma API response to get all necessary information
+ * to call the Figma client and receive the image urls.
+ *
+ * @param {Array<Object>} exportableData
+ * @return {Array<Object>}
+ */
+const parseImageRequestData = (exportableData) =>
+  exportableData.reduce((acc, component) => {
+      component.exportSettings.forEach((setting) => {
+        const { suffix } = setting;
+        const format = setting.format.toLowerCase();
+        let scale = 1;
+        if (setting.constraint && setting.constraint.type && setting.constraint.type === 'SCALE') {
+          scale = setting.constraint.value;
+        }
+
+        const existingElement = acc.find(c => c.format === format && c.scale === scale);
+        if (existingElement) {
+          existingElement.ids.push(component.id);
+        } else {
+          acc.push({
+            ids: [component.id],
```

Comment:
Why would id need to be a tuple instead of just a string?

Reply:
Adding all image asset ids with the same format and scale to the same object, to limit the amount of Figma API requests necessary. The API call for images is not really intended to be used per asset, but per format, with an array of asset ids.